File: src/BackEnd.java
import spec.Language;

public class BackEnd  implements Programmer{
    private Server server;
    private Language language;

    @Override
    public Program makeProgram(Paper paper) {
        paper.setData(this);
//        if(paper instanceof ServerClient) {
//            ServerClient pa = (ServerClient) paper;
//            this.server = pa.server;
//            this.language = pa.backEndLanguage;
//        }
        return makeBackEndProgram();
    }

    private Program makeBackEndProgram() {
        return new Program();
    }

    public void setLanguage(Language backEndLanguage) {
    }

    public void setServer(Server server) {
    }
}

File: Swift/Programmer/Programmer/Example/Programmer/BackEnd.swift
//
//  BackEnd.swift
//  Programmer
//
//  Created by tskim on 19/10/2019.
//  Copyright © 2019 assin. All rights reserved.
//

import Foundation


class BackEnd: Programmer {
    private var server: Server?
    private var language: Language?
    
    func makeProgram(paper: Paper) -> Program {
        /*
         BackEnd 도 마찬가지로 리스코프치환원칙을 위반하고 있다.
         */
        if paper is ServerClient {
            let pb = paper as! ServerClient
            self.server = pb.server
            self.language = pb.backendLanguage
        }
        // else 가 없으면 아무런 에러가 발생하지 않는다. (Context error 가 발생)
        return makeBackEndProgram()
    }
    
    private func makeBackEndProgram() -> Program {
       return Program()
    }
}
